{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "combined-sapphire",
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate bed and wig files for the diff. expressed regions\n",
    "# use results from samtools depth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3a54ed76",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"..\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "06fbd931",
   "metadata": {},
   "outputs": [],
   "source": [
    "outDir = \"Figures-and-Tables/\"\n",
    "sampleAnnotFile = \"annotation/annotation-63.csv\"\n",
    "diffExprValuesFile =  paste0(outDir,\"DGE-results.RData\") # differential expression results\n",
    "diffExprTilesFile = \"differentially-expressed-tiles.bed\" # will be produced for UCSC\n",
    "regionsFile = \"data/signif-regions-extended-for-covg-plot.bed\" # regions around differential expr. results\n",
    "ntWiseCoverageFile = \"data/sam-depth-25MpercIDS-uniqMap-wFlags-all-signif-regions-extended.txt\" \n",
    "  # -> for each nucleotide and sample what's the coverage with reads, from samtool\n",
    "wigOutFileBase = \"signif-region\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cb7279ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>gender</th><th scope=col>age</th><th scope=col>group</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SXR0002</td><td>f</td><td>80</td><td>ccRCC       </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SXR0004</td><td>m</td><td>50</td><td>ccRCC       </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>SXR0006</td><td>m</td><td>68</td><td>ccRCC       </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SXR0010</td><td>m</td><td>65</td><td>urolithiasis</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>SXR0014</td><td>m</td><td>57</td><td>ccRCC       </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>SXR0016</td><td>m</td><td>59</td><td>ccRCC       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & ID & gender & age & group\\\\\n",
       "  & <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & SXR0002 & f & 80 & ccRCC       \\\\\n",
       "\t2 & SXR0004 & m & 50 & ccRCC       \\\\\n",
       "\t3 & SXR0006 & m & 68 & ccRCC       \\\\\n",
       "\t4 & SXR0010 & m & 65 & urolithiasis\\\\\n",
       "\t5 & SXR0014 & m & 57 & ccRCC       \\\\\n",
       "\t6 & SXR0016 & m & 59 & ccRCC       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | gender &lt;chr&gt; | age &lt;int&gt; | group &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | SXR0002 | f | 80 | ccRCC        |\n",
       "| 2 | SXR0004 | m | 50 | ccRCC        |\n",
       "| 3 | SXR0006 | m | 68 | ccRCC        |\n",
       "| 4 | SXR0010 | m | 65 | urolithiasis |\n",
       "| 5 | SXR0014 | m | 57 | ccRCC        |\n",
       "| 6 | SXR0016 | m | 59 | ccRCC        |\n",
       "\n"
      ],
      "text/plain": [
       "  ID      gender age group       \n",
       "1 SXR0002 f      80  ccRCC       \n",
       "2 SXR0004 m      50  ccRCC       \n",
       "3 SXR0006 m      68  ccRCC       \n",
       "4 SXR0010 m      65  urolithiasis\n",
       "5 SXR0014 m      57  ccRCC       \n",
       "6 SXR0016 m      59  ccRCC       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sampleAnnot = read.csv(file = sampleAnnotFile,\n",
    "                        sep = \"\\t\", dec=\".\", stringsAsFactors = F )\n",
    "head(sampleAnnot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dramatic-transsexual",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>SXR0002</dt><dd>'tumor'</dd><dt>SXR0004</dt><dd>'tumor'</dd><dt>SXR0006</dt><dd>'tumor'</dd><dt>SXR0010</dt><dd>'normal'</dd><dt>SXR0014</dt><dd>'tumor'</dd><dt>SXR0016</dt><dd>'tumor'</dd><dt>SXR0018</dt><dd>'normal'</dd><dt>SXR0028</dt><dd>'tumor'</dd><dt>SXR0029</dt><dd>'tumor'</dd><dt>SXR0030</dt><dd>'tumor'</dd><dt>SXR0031</dt><dd>'tumor'</dd><dt>SXR0032</dt><dd>'tumor'</dd><dt>SXR0033</dt><dd>'tumor'</dd><dt>SXR0035</dt><dd>'normal'</dd><dt>SXR0038</dt><dd>'normal'</dd><dt>SXR0039</dt><dd>'tumor'</dd><dt>SXR0040</dt><dd>'tumor'</dd><dt>SXR0042</dt><dd>'tumor'</dd><dt>SXR0043</dt><dd>'tumor'</dd><dt>SXR0044</dt><dd>'tumor'</dd><dt>SXR0045</dt><dd>'tumor'</dd><dt>SXR0047</dt><dd>'normal'</dd><dt>SXR0048</dt><dd>'normal'</dd><dt>SXR0049</dt><dd>'normal'</dd><dt>SXR0051</dt><dd>'tumor'</dd><dt>SXR0052</dt><dd>'tumor'</dd><dt>SXR0053</dt><dd>'tumor'</dd><dt>SXR0057</dt><dd>'tumor'</dd><dt>SXR0060</dt><dd>'normal'</dd><dt>SXR0063</dt><dd>'tumor'</dd><dt>SXR0064</dt><dd>'tumor'</dd><dt>SXR0065</dt><dd>'tumor'</dd><dt>SXR0067</dt><dd>'tumor'</dd><dt>SXR0068</dt><dd>'tumor'</dd><dt>SXR0069</dt><dd>'tumor'</dd><dt>SXR0070</dt><dd>'normal'</dd><dt>SXR0071</dt><dd>'normal'</dd><dt>SXR0075</dt><dd>'tumor'</dd><dt>SXR0076</dt><dd>'tumor'</dd><dt>SXR0077</dt><dd>'tumor'</dd><dt>SXR0078</dt><dd>'tumor'</dd><dt>SXR0080</dt><dd>'tumor'</dd><dt>SXR0084</dt><dd>'normal'</dd><dt>SXR0089</dt><dd>'tumor'</dd><dt>SXR0091</dt><dd>'tumor'</dd><dt>SXR0092</dt><dd>'tumor'</dd><dt>SXR0093</dt><dd>'tumor'</dd><dt>SXR0098</dt><dd>'normal'</dd><dt>SXR0100</dt><dd>'tumor'</dd><dt>SXR0101</dt><dd>'tumor'</dd><dt>SXR0102</dt><dd>'tumor'</dd><dt>SXR0103</dt><dd>'tumor'</dd><dt>SXR0104</dt><dd>'tumor'</dd><dt>SXR0105</dt><dd>'tumor'</dd><dt>SXR0111</dt><dd>'tumor'</dd><dt>SXR0112</dt><dd>'tumor'</dd><dt>SXR0113</dt><dd>'tumor'</dd><dt>SXR0114</dt><dd>'tumor'</dd><dt>SXR0115</dt><dd>'tumor'</dd><dt>SXR0117</dt><dd>'normal'</dd><dt>SXR0118</dt><dd>'normal'</dd><dt>SXR0121</dt><dd>'normal'</dd><dt>SXR0122</dt><dd>'normal'</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[SXR0002] 'tumor'\n",
       "\\item[SXR0004] 'tumor'\n",
       "\\item[SXR0006] 'tumor'\n",
       "\\item[SXR0010] 'normal'\n",
       "\\item[SXR0014] 'tumor'\n",
       "\\item[SXR0016] 'tumor'\n",
       "\\item[SXR0018] 'normal'\n",
       "\\item[SXR0028] 'tumor'\n",
       "\\item[SXR0029] 'tumor'\n",
       "\\item[SXR0030] 'tumor'\n",
       "\\item[SXR0031] 'tumor'\n",
       "\\item[SXR0032] 'tumor'\n",
       "\\item[SXR0033] 'tumor'\n",
       "\\item[SXR0035] 'normal'\n",
       "\\item[SXR0038] 'normal'\n",
       "\\item[SXR0039] 'tumor'\n",
       "\\item[SXR0040] 'tumor'\n",
       "\\item[SXR0042] 'tumor'\n",
       "\\item[SXR0043] 'tumor'\n",
       "\\item[SXR0044] 'tumor'\n",
       "\\item[SXR0045] 'tumor'\n",
       "\\item[SXR0047] 'normal'\n",
       "\\item[SXR0048] 'normal'\n",
       "\\item[SXR0049] 'normal'\n",
       "\\item[SXR0051] 'tumor'\n",
       "\\item[SXR0052] 'tumor'\n",
       "\\item[SXR0053] 'tumor'\n",
       "\\item[SXR0057] 'tumor'\n",
       "\\item[SXR0060] 'normal'\n",
       "\\item[SXR0063] 'tumor'\n",
       "\\item[SXR0064] 'tumor'\n",
       "\\item[SXR0065] 'tumor'\n",
       "\\item[SXR0067] 'tumor'\n",
       "\\item[SXR0068] 'tumor'\n",
       "\\item[SXR0069] 'tumor'\n",
       "\\item[SXR0070] 'normal'\n",
       "\\item[SXR0071] 'normal'\n",
       "\\item[SXR0075] 'tumor'\n",
       "\\item[SXR0076] 'tumor'\n",
       "\\item[SXR0077] 'tumor'\n",
       "\\item[SXR0078] 'tumor'\n",
       "\\item[SXR0080] 'tumor'\n",
       "\\item[SXR0084] 'normal'\n",
       "\\item[SXR0089] 'tumor'\n",
       "\\item[SXR0091] 'tumor'\n",
       "\\item[SXR0092] 'tumor'\n",
       "\\item[SXR0093] 'tumor'\n",
       "\\item[SXR0098] 'normal'\n",
       "\\item[SXR0100] 'tumor'\n",
       "\\item[SXR0101] 'tumor'\n",
       "\\item[SXR0102] 'tumor'\n",
       "\\item[SXR0103] 'tumor'\n",
       "\\item[SXR0104] 'tumor'\n",
       "\\item[SXR0105] 'tumor'\n",
       "\\item[SXR0111] 'tumor'\n",
       "\\item[SXR0112] 'tumor'\n",
       "\\item[SXR0113] 'tumor'\n",
       "\\item[SXR0114] 'tumor'\n",
       "\\item[SXR0115] 'tumor'\n",
       "\\item[SXR0117] 'normal'\n",
       "\\item[SXR0118] 'normal'\n",
       "\\item[SXR0121] 'normal'\n",
       "\\item[SXR0122] 'normal'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "SXR0002\n",
       ":   'tumor'SXR0004\n",
       ":   'tumor'SXR0006\n",
       ":   'tumor'SXR0010\n",
       ":   'normal'SXR0014\n",
       ":   'tumor'SXR0016\n",
       ":   'tumor'SXR0018\n",
       ":   'normal'SXR0028\n",
       ":   'tumor'SXR0029\n",
       ":   'tumor'SXR0030\n",
       ":   'tumor'SXR0031\n",
       ":   'tumor'SXR0032\n",
       ":   'tumor'SXR0033\n",
       ":   'tumor'SXR0035\n",
       ":   'normal'SXR0038\n",
       ":   'normal'SXR0039\n",
       ":   'tumor'SXR0040\n",
       ":   'tumor'SXR0042\n",
       ":   'tumor'SXR0043\n",
       ":   'tumor'SXR0044\n",
       ":   'tumor'SXR0045\n",
       ":   'tumor'SXR0047\n",
       ":   'normal'SXR0048\n",
       ":   'normal'SXR0049\n",
       ":   'normal'SXR0051\n",
       ":   'tumor'SXR0052\n",
       ":   'tumor'SXR0053\n",
       ":   'tumor'SXR0057\n",
       ":   'tumor'SXR0060\n",
       ":   'normal'SXR0063\n",
       ":   'tumor'SXR0064\n",
       ":   'tumor'SXR0065\n",
       ":   'tumor'SXR0067\n",
       ":   'tumor'SXR0068\n",
       ":   'tumor'SXR0069\n",
       ":   'tumor'SXR0070\n",
       ":   'normal'SXR0071\n",
       ":   'normal'SXR0075\n",
       ":   'tumor'SXR0076\n",
       ":   'tumor'SXR0077\n",
       ":   'tumor'SXR0078\n",
       ":   'tumor'SXR0080\n",
       ":   'tumor'SXR0084\n",
       ":   'normal'SXR0089\n",
       ":   'tumor'SXR0091\n",
       ":   'tumor'SXR0092\n",
       ":   'tumor'SXR0093\n",
       ":   'tumor'SXR0098\n",
       ":   'normal'SXR0100\n",
       ":   'tumor'SXR0101\n",
       ":   'tumor'SXR0102\n",
       ":   'tumor'SXR0103\n",
       ":   'tumor'SXR0104\n",
       ":   'tumor'SXR0105\n",
       ":   'tumor'SXR0111\n",
       ":   'tumor'SXR0112\n",
       ":   'tumor'SXR0113\n",
       ":   'tumor'SXR0114\n",
       ":   'tumor'SXR0115\n",
       ":   'tumor'SXR0117\n",
       ":   'normal'SXR0118\n",
       ":   'normal'SXR0121\n",
       ":   'normal'SXR0122\n",
       ":   'normal'\n",
       "\n"
      ],
      "text/plain": [
       " SXR0002  SXR0004  SXR0006  SXR0010  SXR0014  SXR0016  SXR0018  SXR0028 \n",
       " \"tumor\"  \"tumor\"  \"tumor\" \"normal\"  \"tumor\"  \"tumor\" \"normal\"  \"tumor\" \n",
       " SXR0029  SXR0030  SXR0031  SXR0032  SXR0033  SXR0035  SXR0038  SXR0039 \n",
       " \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\" \"normal\" \"normal\"  \"tumor\" \n",
       " SXR0040  SXR0042  SXR0043  SXR0044  SXR0045  SXR0047  SXR0048  SXR0049 \n",
       " \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\" \"normal\" \"normal\" \"normal\" \n",
       " SXR0051  SXR0052  SXR0053  SXR0057  SXR0060  SXR0063  SXR0064  SXR0065 \n",
       " \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\" \"normal\"  \"tumor\"  \"tumor\"  \"tumor\" \n",
       " SXR0067  SXR0068  SXR0069  SXR0070  SXR0071  SXR0075  SXR0076  SXR0077 \n",
       " \"tumor\"  \"tumor\"  \"tumor\" \"normal\" \"normal\"  \"tumor\"  \"tumor\"  \"tumor\" \n",
       " SXR0078  SXR0080  SXR0084  SXR0089  SXR0091  SXR0092  SXR0093  SXR0098 \n",
       " \"tumor\"  \"tumor\" \"normal\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\" \"normal\" \n",
       " SXR0100  SXR0101  SXR0102  SXR0103  SXR0104  SXR0105  SXR0111  SXR0112 \n",
       " \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\"  \"tumor\" \n",
       " SXR0113  SXR0114  SXR0115  SXR0117  SXR0118  SXR0121  SXR0122 \n",
       " \"tumor\"  \"tumor\"  \"tumor\" \"normal\" \"normal\" \"normal\" \"normal\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47  tumorSamples"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SXR0002'</li><li>'SXR0004'</li><li>'SXR0006'</li><li>'SXR0014'</li><li>'SXR0016'</li><li>'SXR0028'</li><li>'SXR0029'</li><li>'SXR0030'</li><li>'SXR0031'</li><li>'SXR0032'</li><li>'SXR0033'</li><li>'SXR0039'</li><li>'SXR0040'</li><li>'SXR0042'</li><li>'SXR0043'</li><li>'SXR0044'</li><li>'SXR0045'</li><li>'SXR0051'</li><li>'SXR0052'</li><li>'SXR0053'</li><li>'SXR0057'</li><li>'SXR0063'</li><li>'SXR0064'</li><li>'SXR0065'</li><li>'SXR0067'</li><li>'SXR0068'</li><li>'SXR0069'</li><li>'SXR0075'</li><li>'SXR0076'</li><li>'SXR0077'</li><li>'SXR0078'</li><li>'SXR0080'</li><li>'SXR0089'</li><li>'SXR0091'</li><li>'SXR0092'</li><li>'SXR0093'</li><li>'SXR0100'</li><li>'SXR0101'</li><li>'SXR0102'</li><li>'SXR0103'</li><li>'SXR0104'</li><li>'SXR0105'</li><li>'SXR0111'</li><li>'SXR0112'</li><li>'SXR0113'</li><li>'SXR0114'</li><li>'SXR0115'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SXR0002'\n",
       "\\item 'SXR0004'\n",
       "\\item 'SXR0006'\n",
       "\\item 'SXR0014'\n",
       "\\item 'SXR0016'\n",
       "\\item 'SXR0028'\n",
       "\\item 'SXR0029'\n",
       "\\item 'SXR0030'\n",
       "\\item 'SXR0031'\n",
       "\\item 'SXR0032'\n",
       "\\item 'SXR0033'\n",
       "\\item 'SXR0039'\n",
       "\\item 'SXR0040'\n",
       "\\item 'SXR0042'\n",
       "\\item 'SXR0043'\n",
       "\\item 'SXR0044'\n",
       "\\item 'SXR0045'\n",
       "\\item 'SXR0051'\n",
       "\\item 'SXR0052'\n",
       "\\item 'SXR0053'\n",
       "\\item 'SXR0057'\n",
       "\\item 'SXR0063'\n",
       "\\item 'SXR0064'\n",
       "\\item 'SXR0065'\n",
       "\\item 'SXR0067'\n",
       "\\item 'SXR0068'\n",
       "\\item 'SXR0069'\n",
       "\\item 'SXR0075'\n",
       "\\item 'SXR0076'\n",
       "\\item 'SXR0077'\n",
       "\\item 'SXR0078'\n",
       "\\item 'SXR0080'\n",
       "\\item 'SXR0089'\n",
       "\\item 'SXR0091'\n",
       "\\item 'SXR0092'\n",
       "\\item 'SXR0093'\n",
       "\\item 'SXR0100'\n",
       "\\item 'SXR0101'\n",
       "\\item 'SXR0102'\n",
       "\\item 'SXR0103'\n",
       "\\item 'SXR0104'\n",
       "\\item 'SXR0105'\n",
       "\\item 'SXR0111'\n",
       "\\item 'SXR0112'\n",
       "\\item 'SXR0113'\n",
       "\\item 'SXR0114'\n",
       "\\item 'SXR0115'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SXR0002'\n",
       "2. 'SXR0004'\n",
       "3. 'SXR0006'\n",
       "4. 'SXR0014'\n",
       "5. 'SXR0016'\n",
       "6. 'SXR0028'\n",
       "7. 'SXR0029'\n",
       "8. 'SXR0030'\n",
       "9. 'SXR0031'\n",
       "10. 'SXR0032'\n",
       "11. 'SXR0033'\n",
       "12. 'SXR0039'\n",
       "13. 'SXR0040'\n",
       "14. 'SXR0042'\n",
       "15. 'SXR0043'\n",
       "16. 'SXR0044'\n",
       "17. 'SXR0045'\n",
       "18. 'SXR0051'\n",
       "19. 'SXR0052'\n",
       "20. 'SXR0053'\n",
       "21. 'SXR0057'\n",
       "22. 'SXR0063'\n",
       "23. 'SXR0064'\n",
       "24. 'SXR0065'\n",
       "25. 'SXR0067'\n",
       "26. 'SXR0068'\n",
       "27. 'SXR0069'\n",
       "28. 'SXR0075'\n",
       "29. 'SXR0076'\n",
       "30. 'SXR0077'\n",
       "31. 'SXR0078'\n",
       "32. 'SXR0080'\n",
       "33. 'SXR0089'\n",
       "34. 'SXR0091'\n",
       "35. 'SXR0092'\n",
       "36. 'SXR0093'\n",
       "37. 'SXR0100'\n",
       "38. 'SXR0101'\n",
       "39. 'SXR0102'\n",
       "40. 'SXR0103'\n",
       "41. 'SXR0104'\n",
       "42. 'SXR0105'\n",
       "43. 'SXR0111'\n",
       "44. 'SXR0112'\n",
       "45. 'SXR0113'\n",
       "46. 'SXR0114'\n",
       "47. 'SXR0115'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SXR0002\" \"SXR0004\" \"SXR0006\" \"SXR0014\" \"SXR0016\" \"SXR0028\" \"SXR0029\"\n",
       " [8] \"SXR0030\" \"SXR0031\" \"SXR0032\" \"SXR0033\" \"SXR0039\" \"SXR0040\" \"SXR0042\"\n",
       "[15] \"SXR0043\" \"SXR0044\" \"SXR0045\" \"SXR0051\" \"SXR0052\" \"SXR0053\" \"SXR0057\"\n",
       "[22] \"SXR0063\" \"SXR0064\" \"SXR0065\" \"SXR0067\" \"SXR0068\" \"SXR0069\" \"SXR0075\"\n",
       "[29] \"SXR0076\" \"SXR0077\" \"SXR0078\" \"SXR0080\" \"SXR0089\" \"SXR0091\" \"SXR0092\"\n",
       "[36] \"SXR0093\" \"SXR0100\" \"SXR0101\" \"SXR0102\" \"SXR0103\" \"SXR0104\" \"SXR0105\"\n",
       "[43] \"SXR0111\" \"SXR0112\" \"SXR0113\" \"SXR0114\" \"SXR0115\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16  normalSamples"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SXR0010'</li><li>'SXR0018'</li><li>'SXR0035'</li><li>'SXR0038'</li><li>'SXR0047'</li><li>'SXR0048'</li><li>'SXR0049'</li><li>'SXR0060'</li><li>'SXR0070'</li><li>'SXR0071'</li><li>'SXR0084'</li><li>'SXR0098'</li><li>'SXR0117'</li><li>'SXR0118'</li><li>'SXR0121'</li><li>'SXR0122'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SXR0010'\n",
       "\\item 'SXR0018'\n",
       "\\item 'SXR0035'\n",
       "\\item 'SXR0038'\n",
       "\\item 'SXR0047'\n",
       "\\item 'SXR0048'\n",
       "\\item 'SXR0049'\n",
       "\\item 'SXR0060'\n",
       "\\item 'SXR0070'\n",
       "\\item 'SXR0071'\n",
       "\\item 'SXR0084'\n",
       "\\item 'SXR0098'\n",
       "\\item 'SXR0117'\n",
       "\\item 'SXR0118'\n",
       "\\item 'SXR0121'\n",
       "\\item 'SXR0122'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SXR0010'\n",
       "2. 'SXR0018'\n",
       "3. 'SXR0035'\n",
       "4. 'SXR0038'\n",
       "5. 'SXR0047'\n",
       "6. 'SXR0048'\n",
       "7. 'SXR0049'\n",
       "8. 'SXR0060'\n",
       "9. 'SXR0070'\n",
       "10. 'SXR0071'\n",
       "11. 'SXR0084'\n",
       "12. 'SXR0098'\n",
       "13. 'SXR0117'\n",
       "14. 'SXR0118'\n",
       "15. 'SXR0121'\n",
       "16. 'SXR0122'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SXR0010\" \"SXR0018\" \"SXR0035\" \"SXR0038\" \"SXR0047\" \"SXR0048\" \"SXR0049\"\n",
       " [8] \"SXR0060\" \"SXR0070\" \"SXR0071\" \"SXR0084\" \"SXR0098\" \"SXR0117\" \"SXR0118\"\n",
       "[15] \"SXR0121\" \"SXR0122\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "group = ifelse(sampleAnnot$group==\"ccRCC\",\"tumor\",\"normal\")\n",
    "names(group) = sampleAnnot$ID\n",
    "group\n",
    "tumorSamples = names(group[ group == \"tumor\"])\n",
    "normalSamples = names(group[ group == \"normal\"])\n",
    "cat(length(tumorSamples),\" tumorSamples\"); tumorSamples\n",
    "cat(length(normalSamples),\" normalSamples\"); normalSamples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "martial-semiconductor",
   "metadata": {},
   "source": [
    "### diff. expressed tiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "42359ac6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 84</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene</th><th scope=col>EnsgID</th><th scope=col>geneType</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>⋯</th><th scope=col>SXR0049</th><th scope=col>SXR0060</th><th scope=col>SXR0070</th><th scope=col>SXR0071</th><th scope=col>SXR0084</th><th scope=col>SXR0098</th><th scope=col>SXR0117</th><th scope=col>SXR0118</th><th scope=col>SXR0121</th><th scope=col>SXR0122</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>chr1_28905201_28905300</th><td>chr1 </td><td> 28905201</td><td> 28905300</td><td>SNORD99 </td><td>ENSG00000197989;ENSG00000221539</td><td>antisense;snoRNA             </td><td> 3.886967</td><td>-3.450529</td><td>0.6894305</td><td>-5.004897</td><td>⋯</td><td> 0</td><td> 7</td><td>23</td><td>11</td><td> 2</td><td> 0</td><td>  23</td><td>1</td><td>   7</td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>chr1_153643701_153643800</th><td>chr1 </td><td>153643701</td><td>153643800</td><td>TRNA_Met</td><td>ENSG00000224870;ENSG00000242485</td><td>protein_coding;protein_coding</td><td>20.566221</td><td>-1.632384</td><td>0.3768256</td><td>-4.331934</td><td>⋯</td><td>25</td><td>58</td><td>33</td><td>76</td><td>34</td><td> 8</td><td>  42</td><td>0</td><td>  24</td><td>34</td></tr>\n",
       "\t<tr><th scope=row>chr1_153643801_153643900</th><td>chr1 </td><td>153643801</td><td>153643900</td><td>        </td><td>ENSG00000224870;ENSG00000242485</td><td>protein_coding;protein_coding</td><td>11.730551</td><td>-2.187097</td><td>0.5535580</td><td>-3.950981</td><td>⋯</td><td> 9</td><td>49</td><td>20</td><td>44</td><td>24</td><td> 3</td><td>  30</td><td>0</td><td>  10</td><td>17</td></tr>\n",
       "\t<tr><th scope=row>chr3_186504601_186504700</th><td>chr3 </td><td>186504601</td><td>186504700</td><td>SNORA81 </td><td>ENSG00000156976;ENSG00000221420</td><td>protein_coding;snoRNA        </td><td> 5.329213</td><td>-1.799555</td><td>0.4319252</td><td>-4.166357</td><td>⋯</td><td> 2</td><td>18</td><td> 7</td><td> 6</td><td>15</td><td> 1</td><td>  29</td><td>0</td><td>   7</td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>chr6_86387301_86387400</th><td>chr6 </td><td> 86387301</td><td> 86387400</td><td>SNORD50B</td><td>ENSG00000203875                </td><td>processed_transcript         </td><td>11.751297</td><td>-2.178942</td><td>0.5842090</td><td>-3.729730</td><td>⋯</td><td> 4</td><td>12</td><td>23</td><td>13</td><td> 1</td><td> 3</td><td>  32</td><td>1</td><td>  21</td><td> 3</td></tr>\n",
       "\t<tr><th scope=row>chr11_62609001_62609100</th><td>chr11</td><td> 62609001</td><td> 62609100</td><td>RNU2-2P </td><td>ENSG00000133316;ENSG00000222328</td><td>protein_coding;snRNA         </td><td>84.285631</td><td>-2.188605</td><td>0.5840148</td><td>-3.747516</td><td>⋯</td><td>11</td><td>59</td><td>19</td><td> 8</td><td>29</td><td>10</td><td>1557</td><td>1</td><td>1148</td><td>21</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 84\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr & start & end & gene & EnsgID & geneType & baseMean & log2FoldChange & lfcSE & stat & ⋯ & SXR0049 & SXR0060 & SXR0070 & SXR0071 & SXR0084 & SXR0098 & SXR0117 & SXR0118 & SXR0121 & SXR0122\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tchr1\\_28905201\\_28905300 & chr1  &  28905201 &  28905300 & SNORD99  & ENSG00000197989;ENSG00000221539 & antisense;snoRNA              &  3.886967 & -3.450529 & 0.6894305 & -5.004897 & ⋯ &  0 &  7 & 23 & 11 &  2 &  0 &   23 & 1 &    7 &  5\\\\\n",
       "\tchr1\\_153643701\\_153643800 & chr1  & 153643701 & 153643800 & TRNA\\_Met & ENSG00000224870;ENSG00000242485 & protein\\_coding;protein\\_coding & 20.566221 & -1.632384 & 0.3768256 & -4.331934 & ⋯ & 25 & 58 & 33 & 76 & 34 &  8 &   42 & 0 &   24 & 34\\\\\n",
       "\tchr1\\_153643801\\_153643900 & chr1  & 153643801 & 153643900 &          & ENSG00000224870;ENSG00000242485 & protein\\_coding;protein\\_coding & 11.730551 & -2.187097 & 0.5535580 & -3.950981 & ⋯ &  9 & 49 & 20 & 44 & 24 &  3 &   30 & 0 &   10 & 17\\\\\n",
       "\tchr3\\_186504601\\_186504700 & chr3  & 186504601 & 186504700 & SNORA81  & ENSG00000156976;ENSG00000221420 & protein\\_coding;snoRNA         &  5.329213 & -1.799555 & 0.4319252 & -4.166357 & ⋯ &  2 & 18 &  7 &  6 & 15 &  1 &   29 & 0 &    7 &  5\\\\\n",
       "\tchr6\\_86387301\\_86387400 & chr6  &  86387301 &  86387400 & SNORD50B & ENSG00000203875                 & processed\\_transcript          & 11.751297 & -2.178942 & 0.5842090 & -3.729730 & ⋯ &  4 & 12 & 23 & 13 &  1 &  3 &   32 & 1 &   21 &  3\\\\\n",
       "\tchr11\\_62609001\\_62609100 & chr11 &  62609001 &  62609100 & RNU2-2P  & ENSG00000133316;ENSG00000222328 & protein\\_coding;snRNA          & 84.285631 & -2.188605 & 0.5840148 & -3.747516 & ⋯ & 11 & 59 & 19 &  8 & 29 & 10 & 1557 & 1 & 1148 & 21\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 84\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;dbl&gt; | end &lt;dbl&gt; | gene &lt;chr&gt; | EnsgID &lt;chr&gt; | geneType &lt;chr&gt; | baseMean &lt;dbl&gt; | log2FoldChange &lt;dbl&gt; | lfcSE &lt;dbl&gt; | stat &lt;dbl&gt; | ⋯ ⋯ | SXR0049 &lt;int&gt; | SXR0060 &lt;int&gt; | SXR0070 &lt;int&gt; | SXR0071 &lt;int&gt; | SXR0084 &lt;int&gt; | SXR0098 &lt;int&gt; | SXR0117 &lt;int&gt; | SXR0118 &lt;int&gt; | SXR0121 &lt;int&gt; | SXR0122 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1_28905201_28905300 | chr1  |  28905201 |  28905300 | SNORD99  | ENSG00000197989;ENSG00000221539 | antisense;snoRNA              |  3.886967 | -3.450529 | 0.6894305 | -5.004897 | ⋯ |  0 |  7 | 23 | 11 |  2 |  0 |   23 | 1 |    7 |  5 |\n",
       "| chr1_153643701_153643800 | chr1  | 153643701 | 153643800 | TRNA_Met | ENSG00000224870;ENSG00000242485 | protein_coding;protein_coding | 20.566221 | -1.632384 | 0.3768256 | -4.331934 | ⋯ | 25 | 58 | 33 | 76 | 34 |  8 |   42 | 0 |   24 | 34 |\n",
       "| chr1_153643801_153643900 | chr1  | 153643801 | 153643900 | <!----> | ENSG00000224870;ENSG00000242485 | protein_coding;protein_coding | 11.730551 | -2.187097 | 0.5535580 | -3.950981 | ⋯ |  9 | 49 | 20 | 44 | 24 |  3 |   30 | 0 |   10 | 17 |\n",
       "| chr3_186504601_186504700 | chr3  | 186504601 | 186504700 | SNORA81  | ENSG00000156976;ENSG00000221420 | protein_coding;snoRNA         |  5.329213 | -1.799555 | 0.4319252 | -4.166357 | ⋯ |  2 | 18 |  7 |  6 | 15 |  1 |   29 | 0 |    7 |  5 |\n",
       "| chr6_86387301_86387400 | chr6  |  86387301 |  86387400 | SNORD50B | ENSG00000203875                 | processed_transcript          | 11.751297 | -2.178942 | 0.5842090 | -3.729730 | ⋯ |  4 | 12 | 23 | 13 |  1 |  3 |   32 | 1 |   21 |  3 |\n",
       "| chr11_62609001_62609100 | chr11 |  62609001 |  62609100 | RNU2-2P  | ENSG00000133316;ENSG00000222328 | protein_coding;snRNA          | 84.285631 | -2.188605 | 0.5840148 | -3.747516 | ⋯ | 11 | 59 | 19 |  8 | 29 | 10 | 1557 | 1 | 1148 | 21 |\n",
       "\n"
      ],
      "text/plain": [
       "                         chr   start     end       gene    \n",
       "chr1_28905201_28905300   chr1   28905201  28905300 SNORD99 \n",
       "chr1_153643701_153643800 chr1  153643701 153643800 TRNA_Met\n",
       "chr1_153643801_153643900 chr1  153643801 153643900         \n",
       "chr3_186504601_186504700 chr3  186504601 186504700 SNORA81 \n",
       "chr6_86387301_86387400   chr6   86387301  86387400 SNORD50B\n",
       "chr11_62609001_62609100  chr11  62609001  62609100 RNU2-2P \n",
       "                         EnsgID                         \n",
       "chr1_28905201_28905300   ENSG00000197989;ENSG00000221539\n",
       "chr1_153643701_153643800 ENSG00000224870;ENSG00000242485\n",
       "chr1_153643801_153643900 ENSG00000224870;ENSG00000242485\n",
       "chr3_186504601_186504700 ENSG00000156976;ENSG00000221420\n",
       "chr6_86387301_86387400   ENSG00000203875                \n",
       "chr11_62609001_62609100  ENSG00000133316;ENSG00000222328\n",
       "                         geneType                      baseMean  log2FoldChange\n",
       "chr1_28905201_28905300   antisense;snoRNA               3.886967 -3.450529     \n",
       "chr1_153643701_153643800 protein_coding;protein_coding 20.566221 -1.632384     \n",
       "chr1_153643801_153643900 protein_coding;protein_coding 11.730551 -2.187097     \n",
       "chr3_186504601_186504700 protein_coding;snoRNA          5.329213 -1.799555     \n",
       "chr6_86387301_86387400   processed_transcript          11.751297 -2.178942     \n",
       "chr11_62609001_62609100  protein_coding;snRNA          84.285631 -2.188605     \n",
       "                         lfcSE     stat      ⋯ SXR0049 SXR0060 SXR0070 SXR0071\n",
       "chr1_28905201_28905300   0.6894305 -5.004897 ⋯  0       7      23      11     \n",
       "chr1_153643701_153643800 0.3768256 -4.331934 ⋯ 25      58      33      76     \n",
       "chr1_153643801_153643900 0.5535580 -3.950981 ⋯  9      49      20      44     \n",
       "chr3_186504601_186504700 0.4319252 -4.166357 ⋯  2      18       7       6     \n",
       "chr6_86387301_86387400   0.5842090 -3.729730 ⋯  4      12      23      13     \n",
       "chr11_62609001_62609100  0.5840148 -3.747516 ⋯ 11      59      19       8     \n",
       "                         SXR0084 SXR0098 SXR0117 SXR0118 SXR0121 SXR0122\n",
       "chr1_28905201_28905300    2       0        23    1          7     5     \n",
       "chr1_153643701_153643800 34       8        42    0         24    34     \n",
       "chr1_153643801_153643900 24       3        30    0         10    17     \n",
       "chr3_186504601_186504700 15       1        29    0          7     5     \n",
       "chr6_86387301_86387400    1       3        32    1         21     3     \n",
       "chr11_62609001_62609100  29      10      1557    1       1148    21     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "load(diffExprValuesFile)\n",
    "head(deseqRes_signif)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fluid-stretch",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for upload into UCSC browser, marks the diff. expressed regions, next to the \n",
    "#  nucleotide-wise coverage produced below:\n",
    "f = file(description = paste0(outDir,diffExprTilesFile), open = \"w\")\n",
    "writeLines(con = f,text = \"track name=\\\"significant regions\\\" description=\\\"differentially expressed regions\\\" color=50,255,50\")\n",
    "write.table(cbind(deseqRes_signif[,c(\"chr\",\"start\",\"end\")],\n",
    "                  \"region\",1000,\"+\",deseqRes_signif[,c(\"start\",\"end\")],\"50,255,50\"), \n",
    "            quote = F,row.names = F,col.names = F, sep=\"\\t\",\n",
    "            file = f)\n",
    "close(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "organized-homework",
   "metadata": {},
   "source": [
    "### load coverage of regions to display"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "respiratory-creativity",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 8 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>descr</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>6</th><td>chr1 </td><td> 28904520</td><td> 28912021</td><td>SNORD99   </td></tr>\n",
       "\t<tr><th scope=row>7</th><td>chr1 </td><td>153643650</td><td>153643999</td><td>tRNA-Met  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr11</td><td> 62599082</td><td> 62618679</td><td>RNU2-2P   </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr11</td><td> 62619136</td><td> 62628032</td><td>SNORD22/26</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr11</td><td> 65264662</td><td> 65282872</td><td>mascRNA   </td></tr>\n",
       "\t<tr><th scope=row>1</th><td>chr17</td><td> 62223674</td><td> 62223919</td><td>SNORA50C  </td></tr>\n",
       "\t<tr><th scope=row>8</th><td>chr3 </td><td>186501100</td><td>186507750</td><td>SNORA81   </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr6 </td><td> 86386590</td><td> 86389928</td><td>SNORD50B  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 8 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & chr & start & end & descr\\\\\n",
       "  & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t6 & chr1  &  28904520 &  28912021 & SNORD99   \\\\\n",
       "\t7 & chr1  & 153643650 & 153643999 & tRNA-Met  \\\\\n",
       "\t2 & chr11 &  62599082 &  62618679 & RNU2-2P   \\\\\n",
       "\t3 & chr11 &  62619136 &  62628032 & SNORD22/26\\\\\n",
       "\t4 & chr11 &  65264662 &  65282872 & mascRNA   \\\\\n",
       "\t1 & chr17 &  62223674 &  62223919 & SNORA50C  \\\\\n",
       "\t8 & chr3  & 186501100 & 186507750 & SNORA81   \\\\\n",
       "\t5 & chr6  &  86386590 &  86389928 & SNORD50B  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 8 × 4\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | descr &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 6 | chr1  |  28904520 |  28912021 | SNORD99    |\n",
       "| 7 | chr1  | 153643650 | 153643999 | tRNA-Met   |\n",
       "| 2 | chr11 |  62599082 |  62618679 | RNU2-2P    |\n",
       "| 3 | chr11 |  62619136 |  62628032 | SNORD22/26 |\n",
       "| 4 | chr11 |  65264662 |  65282872 | mascRNA    |\n",
       "| 1 | chr17 |  62223674 |  62223919 | SNORA50C   |\n",
       "| 8 | chr3  | 186501100 | 186507750 | SNORA81    |\n",
       "| 5 | chr6  |  86386590 |  86389928 | SNORD50B   |\n",
       "\n"
      ],
      "text/plain": [
       "  chr   start     end       descr     \n",
       "6 chr1   28904520  28912021 SNORD99   \n",
       "7 chr1  153643650 153643999 tRNA-Met  \n",
       "2 chr11  62599082  62618679 RNU2-2P   \n",
       "3 chr11  62619136  62628032 SNORD22/26\n",
       "4 chr11  65264662  65282872 mascRNA   \n",
       "1 chr17  62223674  62223919 SNORA50C  \n",
       "8 chr3  186501100 186507750 SNORA81   \n",
       "5 chr6   86386590  86389928 SNORD50B  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "regions  = read.csv(file=regionsFile, header=F, sep=\"\\t\", stringsAsFactors=F, \n",
    "                    col.names = c(\"chr\",\"start\",\"end\",\"descr\"))\n",
    "regions = regions[ order(regions$chr, regions$start, regions$end),]\n",
    "regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6340cced",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 98</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>pos</th><th scope=col>SXR0002</th><th scope=col>SXR0004</th><th scope=col>SXR0006</th><th scope=col>SXR0008</th><th scope=col>SXR0010</th><th scope=col>SXR0012</th><th scope=col>SXR0014</th><th scope=col>SXR0016</th><th scope=col>⋯</th><th scope=col>SXR0112</th><th scope=col>SXR0113</th><th scope=col>SXR0114</th><th scope=col>SXR0115</th><th scope=col>SXR0116</th><th scope=col>SXR0117</th><th scope=col>SXR0118</th><th scope=col>SXR0120</th><th scope=col>SXR0121</th><th scope=col>SXR0122</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>28904521</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>28904522</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>28904523</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 98\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chr & pos & SXR0002 & SXR0004 & SXR0006 & SXR0008 & SXR0010 & SXR0012 & SXR0014 & SXR0016 & ⋯ & SXR0112 & SXR0113 & SXR0114 & SXR0115 & SXR0116 & SXR0117 & SXR0118 & SXR0120 & SXR0121 & SXR0122\\\\\n",
       "  & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 28904521 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t2 & chr1 & 28904522 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t3 & chr1 & 28904523 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 98\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | pos &lt;int&gt; | SXR0002 &lt;int&gt; | SXR0004 &lt;int&gt; | SXR0006 &lt;int&gt; | SXR0008 &lt;int&gt; | SXR0010 &lt;int&gt; | SXR0012 &lt;int&gt; | SXR0014 &lt;int&gt; | SXR0016 &lt;int&gt; | ⋯ ⋯ | SXR0112 &lt;int&gt; | SXR0113 &lt;int&gt; | SXR0114 &lt;int&gt; | SXR0115 &lt;int&gt; | SXR0116 &lt;int&gt; | SXR0117 &lt;int&gt; | SXR0118 &lt;int&gt; | SXR0120 &lt;int&gt; | SXR0121 &lt;int&gt; | SXR0122 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 28904521 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 2 | chr1 | 28904522 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 3 | chr1 | 28904523 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chr  pos      SXR0002 SXR0004 SXR0006 SXR0008 SXR0010 SXR0012 SXR0014 SXR0016\n",
       "1 chr1 28904521 0       0       0       0       0       0       0       0      \n",
       "2 chr1 28904522 0       0       0       0       0       0       0       0      \n",
       "3 chr1 28904523 0       0       0       0       0       0       0       0      \n",
       "  ⋯ SXR0112 SXR0113 SXR0114 SXR0115 SXR0116 SXR0117 SXR0118 SXR0120 SXR0121\n",
       "1 ⋯ 0       0       0       0       0       0       0       0       0      \n",
       "2 ⋯ 0       0       0       0       0       0       0       0       0      \n",
       "3 ⋯ 0       0       0       0       0       0       0       0       0      \n",
       "  SXR0122\n",
       "1 0      \n",
       "2 0      \n",
       "3 0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bedCovgData  = read.csv(file=ntWiseCoverageFile, header=T, sep=\"\\t\", stringsAsFactors=F)\n",
    "colnames(bedCovgData) = gsub(\".Aligned.sortedByCoord.out.25M10percIDS.uniqMap.bam\",\"\",colnames(bedCovgData))\n",
    "colnames(bedCovgData)[1:2] = c(\"chr\",\"pos\")\n",
    "head(bedCovgData,3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "engaged-hormone",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 100</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>mean_coverage_normals</th><th scope=col>mean_coverage_tumors</th><th scope=col>chr</th><th scope=col>pos</th><th scope=col>SXR0002</th><th scope=col>SXR0004</th><th scope=col>SXR0006</th><th scope=col>SXR0008</th><th scope=col>SXR0010</th><th scope=col>SXR0012</th><th scope=col>⋯</th><th scope=col>SXR0112</th><th scope=col>SXR0113</th><th scope=col>SXR0114</th><th scope=col>SXR0115</th><th scope=col>SXR0116</th><th scope=col>SXR0117</th><th scope=col>SXR0118</th><th scope=col>SXR0120</th><th scope=col>SXR0121</th><th scope=col>SXR0122</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>0</td><td>0.0212766</td><td>chr1</td><td>28904521</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>0</td><td>0.0212766</td><td>chr1</td><td>28904522</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>0</td><td>0.0212766</td><td>chr1</td><td>28904523</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 100\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & mean\\_coverage\\_normals & mean\\_coverage\\_tumors & chr & pos & SXR0002 & SXR0004 & SXR0006 & SXR0008 & SXR0010 & SXR0012 & ⋯ & SXR0112 & SXR0113 & SXR0114 & SXR0115 & SXR0116 & SXR0117 & SXR0118 & SXR0120 & SXR0121 & SXR0122\\\\\n",
       "  & <dbl> & <dbl> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 0 & 0.0212766 & chr1 & 28904521 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t2 & 0 & 0.0212766 & chr1 & 28904522 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t3 & 0 & 0.0212766 & chr1 & 28904523 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 100\n",
       "\n",
       "| <!--/--> | mean_coverage_normals &lt;dbl&gt; | mean_coverage_tumors &lt;dbl&gt; | chr &lt;chr&gt; | pos &lt;int&gt; | SXR0002 &lt;int&gt; | SXR0004 &lt;int&gt; | SXR0006 &lt;int&gt; | SXR0008 &lt;int&gt; | SXR0010 &lt;int&gt; | SXR0012 &lt;int&gt; | ⋯ ⋯ | SXR0112 &lt;int&gt; | SXR0113 &lt;int&gt; | SXR0114 &lt;int&gt; | SXR0115 &lt;int&gt; | SXR0116 &lt;int&gt; | SXR0117 &lt;int&gt; | SXR0118 &lt;int&gt; | SXR0120 &lt;int&gt; | SXR0121 &lt;int&gt; | SXR0122 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 0 | 0.0212766 | chr1 | 28904521 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 2 | 0 | 0.0212766 | chr1 | 28904522 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 3 | 0 | 0.0212766 | chr1 | 28904523 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  mean_coverage_normals mean_coverage_tumors chr  pos      SXR0002 SXR0004\n",
       "1 0                     0.0212766            chr1 28904521 0       0      \n",
       "2 0                     0.0212766            chr1 28904522 0       0      \n",
       "3 0                     0.0212766            chr1 28904523 0       0      \n",
       "  SXR0006 SXR0008 SXR0010 SXR0012 ⋯ SXR0112 SXR0113 SXR0114 SXR0115 SXR0116\n",
       "1 0       0       0       0       ⋯ 0       0       0       0       0      \n",
       "2 0       0       0       0       ⋯ 0       0       0       0       0      \n",
       "3 0       0       0       0       ⋯ 0       0       0       0       0      \n",
       "  SXR0117 SXR0118 SXR0120 SXR0121 SXR0122\n",
       "1 0       0       0       0       0      \n",
       "2 0       0       0       0       0      \n",
       "3 0       0       0       0       0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bedCovgData$mean_coverage_normals = rowMeans(bedCovgData[,normalSamples])\n",
    "bedCovgData$mean_coverage_tumors = rowMeans(bedCovgData[,tumorSamples])\n",
    "wh = grep(\"mean\",colnames(bedCovgData))\n",
    "bedCovgData = cbind(bedCovgData[,wh],bedCovgData[,-wh])\n",
    "head(bedCovgData,3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f4a74b0",
   "metadata": {},
   "source": [
    "## export to wig\n",
    " each region 1 file (and 1 track):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "steady-philippines",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1 28904520 28912021 SNORD99 \n",
      "  ->  Figures-and-Tables/signif-region-SNORD99.wig\n",
      "chr1 153643650 153643999 tRNA-Met \n",
      "  ->  Figures-and-Tables/signif-region-tRNA-Met.wig\n",
      "chr11 62599082 62618679 RNU2-2P \n",
      "  ->  Figures-and-Tables/signif-region-RNU2-2P.wig\n",
      "chr11 62619136 62628032 SNORD22/26 \n",
      "  ->  Figures-and-Tables/signif-region-SNORD22_26.wig\n",
      "chr11 65264662 65282872 mascRNA \n",
      "  ->  Figures-and-Tables/signif-region-mascRNA.wig\n",
      "chr17 62223674 62223919 SNORA50C \n",
      "  ->  Figures-and-Tables/signif-region-SNORA50C.wig\n",
      "chr3 186501100 186507750 SNORA81 \n",
      "  ->  Figures-and-Tables/signif-region-SNORA81.wig\n",
      "chr6 86386590 86389928 SNORD50B \n",
      "  ->  Figures-and-Tables/signif-region-SNORD50B.wig\n",
      "\n",
      "\n",
      "regions to copy & paste into UCSC browser:\n",
      "\n",
      "chr1:28904520-28912021  SNORD99\n",
      "chr1:153643650-153643999  tRNA-Met\n",
      "chr11:62599082-62618679  RNU2-2P\n",
      "chr11:62619136-62628032  SNORD22/26\n",
      "chr11:65264662-65282872  mascRNA\n",
      "chr17:62223674-62223919  SNORA50C\n",
      "chr3:186501100-186507750  SNORA81\n",
      "chr6:86386590-86389928  SNORD50B\n"
     ]
    }
   ],
   "source": [
    "regionsCopyPasteUCSC = \"\"\n",
    "\n",
    "for (regIdx in 1:nrow(regions)) {\n",
    "    \n",
    "    cat(regions[regIdx,\"chr\"],regions[regIdx,\"start\"],regions[regIdx,\"end\"],regions[regIdx,\"descr\"],\"\\n\")\n",
    "    \n",
    "    regionsCopyPasteUCSC = \n",
    "        paste0(regionsCopyPasteUCSC,\n",
    "               regions[regIdx,\"chr\"],\":\",\n",
    "               regions[regIdx,\"start\"],\"-\",regions[regIdx,\"end\"],\"  \",\n",
    "               regions[regIdx,\"descr\"],\"\\n\")\n",
    "    \n",
    "    chr = regions[regIdx,\"chr\"]; start = regions[regIdx,\"start\"]; end = regions[regIdx,\"end\"]\n",
    "    wh = which(bedCovgData$chr == chr & bedCovgData$pos >= start & bedCovgData$pos <= end)\n",
    "    bedCovgDataSub = bedCovgData[wh, ]\n",
    "    descr = regions[regIdx,\"descr\"]\n",
    "    descrSimple = gsub(\"[; \\\\/]+\",\"_\",descr)\n",
    "    viewRange = range(c(bedCovgDataSub$mean_coverage_normals, bedCovgDataSub$mean_coverage_tumors))\n",
    "    viewRange[2] = viewRange[2] + 1\n",
    "    viewRange = paste0(round(viewRange),collapse=\":\")\n",
    "    \n",
    "    f = file(description = paste0(outDir,wigOutFileBase,\"-\",descrSimple,\".wig\"),open = \"w\")\n",
    "    \n",
    "    writeLines(con=f, text= paste0(\"browser position \",chr,\":\",start,\"-\",end))\n",
    "\n",
    "    bedCovgDataSub$mean_coverage_normals = round(bedCovgDataSub$mean_coverage_normals)\n",
    "    bedCovgDataSub$mean_coverage_tumors = round(bedCovgDataSub$mean_coverage_tumors)\n",
    "    bedCovgDataSub = bedCovgDataSub[ order(bedCovgDataSub$pos),]\n",
    "    wh = which(duplicated(bedCovgDataSub$pos))\n",
    "    if(length(wh) > 0) {bedCovgDataSub = bedCovgDataSub[-wh,]}\n",
    "\n",
    "    # normal:\n",
    "    wigHeadStr = paste0('track type=wiggle_0 name=\"',descr,\n",
    "                        ' - urolithiasis\" description=\"',descr,\n",
    "                        ' - urolithiasis\" visibility=dense autoScale=off ',\n",
    "            'viewLimits=',viewRange,' color=50,50,255 priority=9\\nvariableStep chrom=',chr)\n",
    "    writeLines(text = wigHeadStr,con = f)\n",
    "    wigLinesOut = apply(bedCovgDataSub[,c(\"pos\",\"mean_coverage_normals\")],1,function(r) paste0(r,collapse=\" \"))\n",
    "    writeLines(text = wigLinesOut,con=f)\n",
    "\n",
    "    # tumor:\n",
    "    wigHeadStr = paste0('track type=wiggle_0 name=\"',descr,\n",
    "                        ' - ccRCC\" description=\"',descr,\n",
    "                        ' - ccRCC\" visibility=dense autoScale=off ',\n",
    "            'viewLimits=',viewRange,' color=255,50,50 priority=9\\nvariableStep chrom=',chr)\n",
    "    writeLines(text = wigHeadStr,con = f)\n",
    "    wigLinesOut = apply(bedCovgDataSub[,c(\"pos\",\"mean_coverage_tumors\")],1,function(r) paste0(r,collapse=\" \"))\n",
    "    writeLines(text = wigLinesOut,con=f)\n",
    "\n",
    "    close(f)\n",
    "                    \n",
    "    cat(\"  -> \",paste0(outDir,wigOutFileBase,\"-\",descrSimple,\".wig\\n\"))\n",
    "\n",
    "}\n",
    "                        \n",
    "cat(\"\\n\\nregions to copy & paste into UCSC browser:\\n\\n\",regionsCopyPasteUCSC, sep=\"\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
